tests/ui/pattern/usefulness/empty-match.exhaustive_patterns.stderr

```diff
@@ -176,9 +176,8 @@ LL |     match_guarded_arm!(0u8);
    = note: match arms with guards don't count towards exhaustivity
 help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
-LL ~                 _ if false => {},
-LL +                 0_u8..=u8::MAX => todo!()
-   |
+LL |                 _ if false => {}, 0_u8..=u8::MAX => todo!()
+   |                                 +++++++++++++++++++++++++++
```

Comment:
This suggestion is *inside* a macro definition. Arguably, we shouldn't emit a suggestion for that, specially if the macro is non-local, but having a span pointing at the relevant `match` would still be useful for macro writers. For now I didn't touch that logic (beyond the different span contexts making the suggestion assume it is in a single line instead of multiple).

Reply:
If anything, the only reasonable suggestion we could provide within a macro is `_ => panic!()`.

Reply:
We now detect the macro case and don't provide a suggestion, only show appropriate spans.